client/state/partner-portal/payment-methods/reducer.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import { AnyAction, Reducer } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { combineReducers } from 'calypso/state/utils';
+import { maskField } from 'calypso/lib/checkout';
+
+const initialState = {
+	fields: {},
+	cardDataComplete: {
+		cardNumber: false,
+		cardCvc: false,
+		cardExpiry: false,
+	},
+	cardDataErrors: {},
+	brand: {},
+};
+
+export const fields: Reducer< Record< string, unknown >, AnyAction > = (
+	state = initialState.fields,
+	action
+) => {
+	switch ( action?.type ) {
+		case 'FIELD_VALUE_SET':
+			return {
+				...state,
+				[ action.payload.key ]: {
+					value: maskField( action.payload.key, state[ action.payload.key ], action.payload.value ),
+					isTouched: true, // mark whether the field has been touched
```

Comment:
My idea with this comment was more "What does it mean when something has been touched?" Like:
`isTouched means that the field have a changed value or has been in a "focus" event state and requires new validation` or something like that (I'm not 100% sure that is actually what happens 😂 ).

The current version doesn't really provide much from a context perspective to enlighten an onboarding developer of what this actually means that we set the `isTouched` state to true 😄